extensions/binding/org.eclipse.smarthome.binding.sonos/src/main/java/org/eclipse/smarthome/binding/sonos/handler/ZonePlayerHandler.java

```diff
@@ -507,6 +511,23 @@ public void onValueReceived(String variable, String value, String service) {
                     updateCurrentURIFormatted(value);
                     break;
                 }
+
+                case "SleepTimerGeneration": {
+                    if (value.equals("0")) {
+                        updateState(new ChannelUID(getThing().getThingTypeUID(), getThing().getUID(), SLEEPTIMER),
```

Comment:
I assume you can use: <code>updateState(SLEEPTIMER), new DecimalType(0));</code>